### PR TITLE
feat(ui): Present items as a hierarchical list

### LINKED ISSUE #335: [UI] Present items as a hierarchical list
With #303 refactoring, we lost the UI for hierarchical item organization.

By convention, item keys behave similarly to AWS S3 keys, that is, the `/` character allows for path component separation.

The user can therefore specifiy keys like `/model_1/data_set_2/seed_3/roc_auc`.

The prefix `/` character is optional, meaning that `/model_1/data_set_2/seed_3/roc_auc` and `model_1/data_set_2/seed_3/roc_auc` and equal.

The goal of this issue is to create mockups to present items in a tree, with  associated features:
- add an item to a view
- Drag and drop an item in a view

**Note 1:** an item key can be the base key for another item. Take for example these two keys:
- `/model_1/plot/roc_auc`, and
- `/model_1/plot`

The item `/model_1/plot` share the base path of `/model_1/plot/roc_auc`. In order to differentiate between the two, a UI suggestion is made:

```
model_1
L plot
    L plot (self)
    L roc_auc
```

The item with special name `plot (self)` designates the item stored at `/model_1/plot`.

Relates to #334 because the tree of items is used to add them to a view.

Edit: Designed screen for reference
Image

### PR DESCRIPTION
**Pull request by rouk1**
A new UI component is introduced: a tree accordion 🪗 

It can have as many nested level as needed and is collapsible.

Transforming a list of keys to tree structure is now done in the report store. It's implemented in the `keysAsTree` function.

1. The function starts by initializing an empty array tree to hold the root nodes of the tree structure.
2. It gets all the keys from the items object (or an empty object if items is null).
3. The function then iterates over each key:
  a. It splits the key into segments, filtering out any empty segments.
  b. It identifies the root segment (first part of the path).
  c. It checks if a node for this root segment already exists in the tree. If not, it creates a new node and adds it to the tree.
  d. It then iterates through the remaining segments of the key:
     - For each segment, it builds the full path up to that point.
     - It checks if a child node with this path already exists.
     - If not, it creates a new child node and adds it to the current node's children.
     - It then moves to this child node for the next iteration.
4. After building the basic structure, it defines an addSelf function:
  - This function recursively goes through the tree.
  - If a node's name matches a key in the original list and it has children, it adds a "(self)" node as the first child.
  - This handles cases where a path is both a key itself and has child keys.
5. Finally, it applies the addSelf function to each root node in the tree.
6. The function returns the completed tree structure.

Usage:

```vue
<script setup lang="ts">
import TreeAccordion, { type TreeAccordionNode } from "@/components/TreeAccordion.vue";

const fileTreeNodes: TreeAccordionNode[] = [
  {
    name: "fraud",
    children: [{ name: "fraud/accuracy" }, { name: "fraud/accuracy3" }],
  },
  {
    name: "fraud2",
    children: [{ name: "fraud2/accuracy" }, { name: "fraud2/accuracy3" }],
  },
  {
    name: "nested",
    children: [
      {
        name: "nested/fraud2/accuracy",
        children: [
          { name: "nested/fraud2/accuracy/self" },
          { name: "nested/fraud2/accuracy/self2" },
          { name: "nested/fraud2/accuracy/self3" },
        ],
      },
      { name: "nested/fraud2/accuracy3", children: [] },
    ],
  },
];
</script>

<template>
  <TreeAccordion :nodes="fileTreeNodes" />
</template
```

https://github.com/user-attachments/assets/b5d83778-1c40-429d-beec-1f20d6814dd9

Fixes #335

### CODE REVIEW

**Review by thomass-dev — APPROVED** [review]

File: frontend/src/components/TreeAccordion.vue [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<script lang="ts">
+export interface TreeAccordionNode {
+  name: string;
+  children?: TreeAccordionNode[];
+  indentationLevel?: number;
```
[the remark this answers is not in the document]
Comment (rouk1) [reply frontend/src/components/TreeAccordion.vue]: Nice catch we now only have a `isRoot` prop.

File: frontend/src/stores/report.ts [hunk 2]
```diff
@@ -133,6 +139,78 @@ export const useReportStore = defineStore("reports", () => {
...
+    const tree: TreeNode[] = [];
+    const keys = Object.keys(items.value || {});
+    for (const key of keys) {
+      const segments = key.split("/").filter((s) => s.length > 0);
+      const rootSegment = segments[0];
+      let currentNode = tree.find((n) => n.name == rootSegment);
+      if (!currentNode) {
+        currentNode = { name: rootSegment, children: [] };
+        tree.push(currentNode);
+      }
```
[the remark this answers is not in the document]
Comment (rouk1) [reply frontend/src/stores/report.ts]: As proposed nodes are now stored in a lookup table for optimization purpose.

**Review by thomass-dev — DISMISSED** [review]
Nice improvement with LUT, thanks!

**Review by augustebaum — APPROVED** [review]
Thanks for the rewrite of the double-click mechanic!

**Pull request merged**